renderdoc/driver/ihv/amd/amd_counters.cpp

```diff
@@ -110,13 +109,41 @@ bool AMDCounters::Init(ApiType apiType, void *pContext)
     return false;
   }
 
+  bool disableCounters = false;
+
+#ifndef RELEASE
```

Comment:
This check would be better as `#if ENABLED(RDOC_DEVEL)`, but I'd rather not assume that !release == debuglayer enabled, as someone could flip on the debug layer during release for testing purposes and run into problems, or want to toggle it off to debug the counters.

In `d3d12_hooks.cpp` there's the logic for enabling or disabling the debug layer before creating the device, you could pass that bool to the `WrappedID3D12Device` constructor and store it. It would then make sense to check for that in `d3d12_replay.cpp` where the wrapped device is available - the current `if(m_Vendor == GPUVendor::AMD)` could become `if(m_Vendor == GPUVendor::AMD && !m_pDevice->DebugLayerEnabled())` or something.

Reply:
This code is to disable AMD hardware counters in debug builds is meant only for "GPA v3.0", so I think it's not good idea to use the if(m_Vendor == GPUVendor::AMD && !m_pDevice->DebugLayerEnabled()) approach - version specific logic will leak out of this file. Perhaps I will change to #if ENABLED(RDOC_DEVEL) or you have any other suggestion?

Reply:
I understand the reason for the check, but I'm not sure I understand what you mean. There's no meaningful version check because once this change is merged, then only the released version of 3.0 will work. The preview of 3.0 won't work and neither will any older versions.

Reply:
Yes Baldur, this change won't work with "v3.0 preview" nor any older version. Though this will ensure that it will work with released v3.0 and subsequent releases i.e. v3.1 and later.